array_functions/03_filter.js

```diff
@@ -18,11 +18,15 @@
 */
 
 function filter(array, callback){
-  var arrayLength = array.length;
-	var newArray = [];
-
-
-  return newArray;
+  let result=[]
+  for (let i = 0; i < array.length; i++) {
+    let isFound = callback(array[i], i, array)
```

Comment:
Excelente que guardes el resultado en una variable.
Como dato, si vas a usar ES6 puedes usar `const` en vez de let excepto para el iterador del for, todos los otros valores no son reasignados.